reconcile/terraform_tgw_attachments.py

```diff
@@ -162,13 +218,52 @@ def run(
     if print_to_file:
         sys.exit()
 
-    aws_api = AWSApi(1, accounts, settings=settings, init_users=False)
+    return working_dirs
+
+
+@defer
+def run(
+    dry_run: bool,
+    print_to_file: Optional[str] = None,
+    enable_deletion: bool = False,
+    thread_pool_size: int = 10,
+    defer: Optional[Callable] = None,
+):
+    settings = queries.get_secret_reader_settings()
+    clusters = queries.get_clusters_with_peering_settings()
+    ocm_map = _build_ocm_map(clusters, settings)
+    accounts = queries.get_aws_accounts(terraform_state=True, ecrs=False)
+
+    # Fetch desired state for cluster-to-vpc(account) VPCs
+    with AWSApi(1, accounts, settings=settings, init_users=False) as awsapi:
```

Comment:
Lets use `reconcile.utils.secret_reader.create_secret_reader()` and pass in `secret_reader` instead - this API class supports it. Passing in settings is the old way.